File: 929. Unique Email Addresses.java
class Solution {
    public int numUniqueEmails(String[] emails) {
        HashSet<String> set = new HashSet<>();
        for (String e : emails) {
            String[] ep = e.split("@");
            StringBuilder sb = new StringBuilder();
            for (int i=0; i<ep[0].length(); i++) {
                if (ep[0].charAt(i) == '.') {
                    continue;
                } else if (ep[0].charAt(i) == '+') {
                    break;
                } else {
                    sb.append(ep[0].charAt(i));
                }
            }
            set.add(sb.toString() + "@" + ep[1]);
        }
        return set.size();
    }
}

File: 929. Unique Email Addresses.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        ans = set()
        for email in emails:
            em = email.split('@')
            fp = "".join(em[0].split('+')[0].split('.'))
            ans.add(fp+"@"+em[1])
        return len(ans)